main.go

```diff
@@ -0,0 +1,151 @@
+package main
+
+// Users will give us bulk correction files following https://tools.ietf.org/html/draft-google-self-published-geofeeds-02
+// It isn't uncommon for the corrections they list to either match what we already have
+// or to be worse. This script can help us work that out. Right now, it only looks
+// at the ISO country code, but checking more fields should be easy enough.
+import (
+	"encoding/csv"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+
+	geoip2 "github.com/oschwald/geoip2-golang"
+)
+
+// Usage Example:
+// go run go/check_csv_corrections/main.go /path/to/corrections.csv /path/to/mmdbfile.mmdb
+func main() {
+	geofeedFilename, mmdbFilename, err := parseArgs()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	db, err := geoip2.Open(mmdbFilename)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		if err := db.Close(); err != nil {
+			log.Println(err)
+		}
+	}()
+
+	totalCount, correctionCount := 0, 0
+	geofeedFH, err := os.Open(geofeedFilename) //nolint: gosec // linter doesn't realize we are cleaning the filepath
+	if err != nil {
+		log.Fatal(err)
+	}
+	csvReader := csv.NewReader(geofeedFH)
+	csvReader.ReuseRecord = true
+	csvReader.Comment = '#'
+	csvReader.FieldsPerRecord = 5
+	csvReader.TrimLeadingSpace = true
+	defer func() {
+		if err := geofeedFH.Close(); err != nil {
+			log.Println(err)
+		}
+	}()
+
+	for {
+		row, err := csvReader.Read()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			geofeedFH.Close() //nolint: gosec, errcheck
+			log.Fatal(err)
+		}
+		totalCount++
+		currentCorrectionCount, err := verifyCorrection(row, db)
+		if err != nil {
+			geofeedFH.Close() //nolint: gosec, errcheck
+			log.Fatal(err)
+		}
+		correctionCount += currentCorrectionCount
+	}
+	if err != nil && err != io.EOF {
+		log.Fatalf("Failed to read file: %s", err)
+	}
+	fmt.Printf(
+		"\nOut of %d potential corrections, %d may be different than our current mappings\n",
+		totalCount,
+		correctionCount,
+	)
+}
+
+func verifyCorrection(correction []string, db *geoip2.Reader) (int, error) {
+	/*
+	   0: network (CIDR or single IP)
+	   1: ISO-3166 country code
+	   2: ISO-3166-2 region code
+	   3: city name
+	   4: postal code
+	*/
+	networkOrIP := correction[0]
+	if !(strings.Contains(networkOrIP, "/")) {
+		if strings.Contains(networkOrIP, ":") {
```

Comment:
Ok, so I made a comment initially to KP that I didn't like this, and thought that we should use `net.ParseIP` and then call `ip.To4()` and check if it was nil to see if we had an IPv6 IP.

But Kevin pointed out that when someone puts in an IPv6 representation of an IPv4 address we actually _want to treat it like an IPv6 address_, not an IPv4 address.

So I think this is doing the right thing.